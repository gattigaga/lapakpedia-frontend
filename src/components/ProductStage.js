import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

import Product from "components/Product";

const Container = styled.div`
  margin-bottom: 32px;
`;

const Header = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 24px;
`;

const Label = styled.span`
  font-size: 24px;
  font-family: Roboto;
`;

const StyledLink = styled(Link)`
  color: #bbb;
  font-size: 14px;
  font-family: Roboto;
  text-decoration: none;
`;

const Content = styled.div`
  display: flex;
`;

const Column = styled.div`
  flex: 1;
  box-sizing: border-box;
  padding-left: ${props => (props.position === "left" ? "0px" : "32px")};
  padding-right: ${props => (props.position === "right" ? "0px" : "32px")};
  margin-bottom: 32px;
`;

const ProductStage = ({ products, label, link }) => (
  <Container>
    <Header>
      <Label>{label}</Label>
      <StyledLink to={link}>View All ></StyledLink>
    </Header>
    <Content>
      {products.map((product, index) => {
        const productsEachRow = 4;
        let position = "center";

        if (index === 0) {
          position = "left";
        } else if (index === productsEachRow - 1) {
          position = "right";
        }

        return (
          <Column key={product._id} position={position}>
            <Product {...product} href="#" />
          </Column>
        );
      })}
    </Content>
  </Container>
);

ProductStage.propTypes = {
  products: PropTypes.array.isRequired,
  label: PropTypes.string,
  link: PropTypes.string
};

ProductStage.defaultProps = {
  label: "Label",
  link: "#"
};

export default ProductStage;
