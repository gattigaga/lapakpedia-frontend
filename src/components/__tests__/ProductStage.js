import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import uuid from "uuid/v4";

import ProductStage from "components/ProductStage";

describe("ProductStage", () => {
  jest.mock("components/Product", () => "Product");

  const setup = propOverrides => {
    const product = {
      _id: uuid(),
      image: require("assets/images/dummy240x240.png"),
      name: "MacBook Pro",
      category: "Technology",
      price: 2800,
      href: "#",
      rating: 4,
      totalReviews: 4
    };

    const props = {
      link: "https://lapakpedia.com/products",
      label: "New Arrival",
      products: [...Array(4)].map(() => product),
      ...propOverrides
    };

    const wrapper = shallow(<ProductStage {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
