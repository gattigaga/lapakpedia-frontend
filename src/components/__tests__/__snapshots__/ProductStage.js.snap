// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductStage should renders default 1`] = `
<styled.div>
  <styled.header>
    <styled.span>
      New Arrival
    </styled.span>
    <Styled(Link)
      to="https://lapakpedia.com/products"
    >
      View All &gt;
    </Styled(Link)>
  </styled.header>
  <styled.div>
    <styled.div
      key="72d4dc1b-2238-4550-b3fa-0d589fa1047f"
      position="left"
    >
      <Product
        _id="72d4dc1b-2238-4550-b3fa-0d589fa1047f"
        category="Technology"
        href="#"
        image="dummy240x240.png"
        name="MacBook Pro"
        price={2800}
        rating={4}
        totalReviews={4}
      />
    </styled.div>
    <styled.div
      key="72d4dc1b-2238-4550-b3fa-0d589fa1047f"
      position="center"
    >
      <Product
        _id="72d4dc1b-2238-4550-b3fa-0d589fa1047f"
        category="Technology"
        href="#"
        image="dummy240x240.png"
        name="MacBook Pro"
        price={2800}
        rating={4}
        totalReviews={4}
      />
    </styled.div>
    <styled.div
      key="72d4dc1b-2238-4550-b3fa-0d589fa1047f"
      position="center"
    >
      <Product
        _id="72d4dc1b-2238-4550-b3fa-0d589fa1047f"
        category="Technology"
        href="#"
        image="dummy240x240.png"
        name="MacBook Pro"
        price={2800}
        rating={4}
        totalReviews={4}
      />
    </styled.div>
    <styled.div
      key="72d4dc1b-2238-4550-b3fa-0d589fa1047f"
      position="right"
    >
      <Product
        _id="72d4dc1b-2238-4550-b3fa-0d589fa1047f"
        category="Technology"
        href="#"
        image="dummy240x240.png"
        name="MacBook Pro"
        price={2800}
        rating={4}
        totalReviews={4}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
